src/Swagger/Serializer/ApiGatewayNormalizer.php

```diff
@@ -94,18 +96,18 @@ public function normalize($object, $format = null, array $context = [])
                     unset($data['definitions'][$definition]['properties'][$property]['readOnly']);
                 }
                 if (isset($propertyOptions['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $propertyOptions['$ref'])) {
-                    $data['definitions'][$definition]['properties'][$property]['$ref'] = str_replace(['-', '_'], '', $propertyOptions['$ref']);
+                    $data['definitions'][$definition]['properties'][$property]['$ref'] = str_replace(['-', ':', '_'], '', $propertyOptions['$ref']);
                 }
                 if (isset($propertyOptions['items']['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $propertyOptions['items']['$ref'])) {
-                    $data['definitions'][$definition]['properties'][$property]['items']['$ref'] = str_replace(['-', '_'], '', $propertyOptions['items']['$ref']);
+                    $data['definitions'][$definition]['properties'][$property]['items']['$ref'] = str_replace(['-', ':', '_'], '', $propertyOptions['items']['$ref']);
                 }
             }
         }
 
         // $data['definitions'] is an instance of \ArrayObject
         foreach (array_keys($data['definitions']->getArrayCopy()) as $definition) {
             if (!preg_match('/^[A-z]+$/', (string) $definition)) {
```

Comment:
This pattern should also be modified to match: `/^[\dA-z]+$/`

Reply:
$ref starts with #/ which would need the check you mentioned above.

Reply:
This is not a `$ref`. This is the definition / model name.